tasmota/xsns_62_esp32_mi_ble.ino

```diff
@@ -2583,6 +2746,27 @@ void MI32GetOneSensorJson(int slot, int hidename){
         }
       }
     }
+    if (p->feature.scale){
+      if(p->eventType.scale || !MI32.mode.triggeredTele || MI32.option.allwaysAggregate
+#ifdef USE_HOME_ASSISTANT
+          ||(hass_mode==2)
+#endif //USE_HOME_ASSISTANT
+      ){
+        if (MI32.option.directBridgeMode) {
+          ResponseAppend_P(PSTR(",\"weight_removed\":%u"), p->weight_removed);
+          ResponseAppend_P(PSTR(",\"weight_stabilized\":%u"), p->weight_stabilized);
+        }
+        ResponseAppend_P(PSTR(",\"weight_unit\":\"%s\""), p->weight_unit);
```

Comment:
Can you check how it works with SetOption19 enambled?
On Esp8266 sensor name limited by 30 chars and looks like
"975C0F_MISCALEV2-de30d1_weigh"
So sometimes sensor with that name has wrong val_tpl value ("..weight_unit" instead of "weight" ).
Seems because shorten to 30 chars sensors names the same for
"975C0F_MISCALEV2-de30d1_weight"
"975C0F_MISCALEV2-de30d1_weight_unit"
I had to use MSCALE2 instead of MISCALEV2 and "wght_unit" to get unique value for sensor names (on esp8266+hm10. May be esp32 have no such limits)

Reply:
This does not happen with ESP32. Concating strings seems to be handled differently here. I changed "weight" to a very long name and it worked fine and also appeared correctly in Home Assistant

`{
  "availability": [],
  ...
  "json_attr_t": "tele/tasmota_ble/MISCALEV1123456",
  "name": "MISCALEV1123456_weight_weight_weight_weight_weight",
  "state_topic": "tele/tasmota_ble/MISCALEV1123456",
  "uniq_id": "MISCALEV1123456_weight_weight_weight_weight_weight",
  "unit_of_meas": "kg",
  "val_tpl": "{{ value_json.weight_weight_weight_weight_weight }}"
}`

To get SetOption19 to work i had to enable USE_HOME_ASSISTANT and disable USE_TASMOTA_DISCOVERY in tasmota_configurations_ESP32.h. I also had to extend the "classes" array in MI32DiscoveryOneMISensor() to be able to get Weight as an Entity otherwise only RSSI was shown. 

My new commit includes the MI32DiscoveryOneMISensor changes. I also removed the underscore from the Json property names ("weight_removed" -> "WeightRemoved"). This matches the names from the other properties better and it saves 1 byte.

IMO shortening names to something not understandable is not really a good solution. Can you get rid of the "975C0F_", i think this is the ESP ID? Also, why not just increasing the string max size to 40?

Reply:
975C0F is a part of MAC.  30 chars is defined in xdrv_12_home_assistant.ino
I won't change this part - it will affect to  most esp8266 drivers)

```
  char unique_id[30];
...
  snprintf_P(unique_id, sizeof(unique_id), PSTR("%06X_%s_%s"), ESP_getChipId(), sensorname, subname);

```

For MI32 BLE i see this part implemented separately.